src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -755,5 +755,19 @@ internal void OnProjectOptionsChanged(ProjectId projectId)
                 RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectChanged, oldSolution, newSolution, projectId);
             }
         }
+
+        /// <summary>
+        /// Update the solution as a result of option changes.
+        /// </summary>
+        internal void OnOptionsChanged()
+        {
+            using (_serializationLock.DisposableWait())
+            {
+                var oldSolution = CurrentSolution;
+                var newSolution = this.SetCurrentSolution(oldSolution.WithOptionsChanged());
+
+                RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectChanged, oldSolution, newSolution);
```

Comment:
why option change is project change? shouldn't it be solution change? and depends on whether we consider option has part of solution snapshot, we should either change solution or not change snapshot.

Reply:
anyway, this won't affect solution crawler (hence diagnostic analyzer) since solution.GetChanges() will return no change.

Reply:
if we want this, we should put Option in SolutionChanges as one that got changed.